mender/templates/api-gateway-deploy.yaml

```diff
@@ -42,11 +42,17 @@ spec:
       {{- with .Values.api_gateway.affinity }}
       affinity: {{ tpl (toYaml .) $ | nindent 8 }}
       {{- end }}
+{{- if .Values.api_gateway.podSecurityContext.enabled }}
```

Comment:
Suppose the case where the Security Context is enable but still the httpPort is 80 or 443 (privileged ports): the helm release won't work.

I'd suggest to set the following check on top of the `api-gateway-deploy.yaml` file: the helm deployment won't start with this setting.

```
{{- if or .Values.api_gateway.podSecurityContext.enabled .Values.api_gateway.containerSecurityContext.enabled }}
  {{- if lt (int .Values.api_gateway.httpPort) 1024  }}
    {{- fail ".Values.api_gateway.httpPort can't be less than 1024 when Security Context is enabled" }}
  {{- end }}
{{- end }}
```

Reply:
I added checks for both the http and https port as suggested.